### PR TITLE
WIP - Update documentation

### PR DESCRIPTION
**Pull request by pmuens**
Uses Rust doc formatting so that the code is commented and documented at the same time and docs can be easily generated.

**NOTE:** Some of the doc comments are vague and some might be incorrect 🙈. Still in the progress of getting familiar with the codebase.

### CODE REVIEW

**Comment by pmuens** [comment]
Tests are failing due to wrong example code in docs. Looking into it tomorrow morning!

**Comment by pmuens** [comment]
Temporarily removed the examples from the docs (Amend the old commit). Will re-add them later on!

File: src/tree.rs [hunk 1]
```diff
@@ -8,13 +8,15 @@ use super::*;
 
 const FANOUT: usize = 2;
 
+/// Tree implementation which consists of `Page`s.
```
Comment (spacejam) [inline src/tree.rs]: Lock-free `Tree` implementation based on a B-Link design (Lehman 81).

File: src/tree.rs [hunk 2]
```diff
@@ -437,13 +450,15 @@ pub enum Data {
 }
 
 impl Data {
+    /// Length of the data stored in `Tree`.
```
Comment (spacejam) [inline src/tree.rs]: stored in the `Page`

File: src/tree.rs [hunk 3]
```diff
@@ -475,7 +491,7 @@ impl Data {
     }
 }
 
-
+/// `Tree` `Node` which stores the actual `Data` and metadata about itself and nearby `Node`s.
```
Comment (spacejam) [inline src/tree.rs]: `Tree` `Node` which is either an index or leaf, and has a pointer to the right sibling if there is one.

**Comment by spacejam** [comment]
closing for now, since a bunch of stuff has changed, but in general more docs are always good!

**Comment by pmuens** [comment]
> closing for now, since a bunch of stuff has changed, but in general more docs are always good!

Yes, I agree. this one got quite stale and you pushed a couple of doc updates the other day. However thanks again for taking the time to look over this 👍

I'll add docs if I see that something is missing while working on the corresponding code.

**Pull request closed without merging**